client/src/components/Profiles.js

```diff
@@ -1,43 +1,91 @@
 import React, { useState, useEffect } from "react";
-import { Container, Message, Search, Table, Button } from "semantic-ui-react";
+import {
+  Message,
+  Search,
+  Table,
+  Button,
+  Grid,
+  Dropdown,
+  Input,
+  Container
+} from "semantic-ui-react";
 import { Link } from "react-router-dom";
-import { getProfiles } from "../api/profiles";
+import { getProfiles, getNationalities } from "../api/profiles";
 
 const Profiles = () => {
   const [profiles, setProfiles] = useState([]);
+  const [nationality, setNationality] = useState([]);
   const [searchInput, setSearchInput] = useState("");
+  const [selectDropdown, setSelectDropdown] = useState([]);
 
   const handleSearchChange = event => {
     setSearchInput(event.target.value);
   };
+  const handleOnChange = (e, data) => {
+    setSelectDropdown(data.value);
+  };
 
   useEffect(() => {
     getProfiles().then(response => {
       setProfiles(response);
     });
+    getNationalities().then(response => {
+      setNationality(response);
+    });
   }, []);
 
+  const options = nationality.map(national => ({
+    key: national.id,
+    text: `${national.id} -${national.nationality} `,
+    value: national.id
+  }));
+
   const searchString = searchInput.toLowerCase();
   const isMatch = field => field.toLowerCase().includes(searchString);
 
-  let filteredProfiles = profiles.filter(
-    profile =>
-      isMatch(profile.first_name) ||
-      isMatch(profile.last_name) ||
-      isMatch(profile.email) ||
-      (profile.address && isMatch(profile.address)) ||
-      (profile.phone_number && isMatch(profile.phone_number)) ||
-      (profile.occupation && isMatch(profile.occupation))
-  );
+  let filterProfiles = () => {
+    return selectDropdown.length === 0
+      ? profiles.filter(
+          profile =>
+            isMatch(profile.first_name) ||
+            isMatch(profile.last_name) ||
+            isMatch(profile.email) ||
+            (profile.address && isMatch(profile.address)) ||
+            (profile.phone_number && isMatch(profile.phone_number)) ||
+            (profile.occupation && isMatch(profile.occupation))
+        )
+      : profiles.filter(profile =>
+          selectDropdown.includes(profile.nationality_id)
+        );
+  };
+
+  let filteredProfiles = filterProfiles();
```

Comment:
I wouldn't make the text filtering conditional on whether there is a selection. Just do both. Otherwise you might select a nationality, then start typing to reduce the matches, but nothing would happen. So I'd do

```
let filteredProfiles = profiles;

if (selectDropdown.length > 0) {
   filteredProfiles = filteredProfiles.filter(profile => selectDropdown.includes(profile.nationality_id));
}

filteredProfiles = filteredProfiles.filter( ... do text matching as before)
```